src/lib.rs

```diff
@@ -89,94 +119,207 @@ impl SecretId {
-            .map_err(|DecryptBytesError::DecryptVerify| DecryptError::DecryptVerify)?;
-        serialisation::deserialise(&bytes).map_err(DecryptError::Deserialization)
+        Ok(deserialise(&self.decrypt_bytes(ciphertext)?)?)
     }
 }
 
 quick_error! {
     #[derive(Debug)]
-    pub enum EncryptError {
-        Serialization(e: SerialisationError) {
-            description("error serializing message")
-            display("error serializing message: {}", e)
+    pub enum EncryptionError {
+        Serialisation(e: SerialisationError) {
+            description("error serialising or deserialising message")
+            display("error serialising or deserialising message: {}", e)
             cause(e)
+            from()
+        }
+        DecryptVerify(_e: ()) {
+            description("error decrypting/verifying message")
+            from()
         }
     }
 }
 
-quick_error! {
-    #[derive(Debug)]
-    pub enum DecryptError {
-        DecryptVerify {
-            description("error decrypting/verifying message")
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn anonymous_bytes_cipher() {
+        let data = b"Humpty Dumpty sat on a wall";
+        let sk = SecretId::new();
+        let sk2 = SecretId::new();
+        let pk = sk.public_id();
+
+        let ciphertext = pk.encrypt_anonymous_bytes(data);
+        assert!(&ciphertext != data);
+
+        let error_res: Result<_, _> = sk.decrypt_anonymous_bytes(data);
+        match error_res {
+            Err(_e) => (),
+            Ok(_) => {
+                panic!("Unexpected result: we're using wrong data, it should have returned error")
+            }
         }
-        Deserialization(e: SerialisationError) {
-            description("error deserializing decrypted message")
-            display("error deserializing decrypted message: {}", e)
-            cause(e)
+
+        let error_res: Result<_, _> = sk2.decrypt_anonymous_bytes(&ciphertext);
+        match error_res {
+            Err(_e) => (),
+            Ok(_) => {
+                panic!("Unexpected result: we're using a wrong key, it should have returned error")
+            }
         }
+
+        let plaintext: Vec<u8> = sk
+            .decrypt_anonymous_bytes(&ciphertext)
+            .expect("couldn't decrypt ciphertext");
+        assert!(&plaintext == data);
     }
-}
 
-quick_error! {
-    #[derive(Clone, Debug)]
-    pub enum DecryptBytesError {
-        DecryptVerify {
-            description("error decrypting/verifying message")
+    #[test]
+    fn anonymous_cipher() {
+        let data: Vec<u64> = vec![4, 5, 6];
+
+        let sk = SecretId::new();
+        let pk = sk.public_id();
+
+        let ciphertext = pk
+            .encrypt_anonymous(&data)
+            .expect("couldn't encrypt base data");
+        assert!(!ciphertext.is_empty());
+
+        let plaintext: Vec<u64> = sk
+            .decrypt_anonymous(&ciphertext)
+            .expect("couldn't decrypt ciphertext");
```

Comment:
why are we using this and depending on strings to debug locate stuff, instead of `unwrap!()` like we do due to our convention